odoo/addons/base/models/ir_model.py

```diff
@@ -286,7 +286,9 @@ def _unlink_if_manual(self):
 
     def unlink(self):
         # prevent screwing up fields that depend on these models' fields
-        self.field_id._prepare_update()
+        manual_models = self.filtered(lambda model: model.state == 'manual')
+        manual_models.field_id.filtered(lambda f: f.state == 'manual')._prepare_update()
```

Comment:
how can you have manual models with fields that are not manual?

Reply:
it is explained in the commit message: when a model implements a mixin (usually mail.thread)